src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -563,27 +573,341 @@ internal SafeFileHandle OpenPort(string portName)
                 _handle = null;
                 throw;
             }
+
+            _processReadDelegate = ProcessRead;
+            _processWriteDelegate = ProcessWrite;
+            _lastTotalBytesAvailable = TotalBytesAvailable;
+        }
+
+        private void EnsureIOLoopRunning()
+        {
+            lock (_ioLoopLock)
+            {
+                if (_ioLoop == null)
+                {
+                    Debug.Assert(_handle != null);
+                    _ioLoop = Task.Factory.StartNew(
+                        IOLoop,
+                        CancellationToken.None,
+                        TaskCreationOptions.LongRunning,
+                        TaskScheduler.Default);
+                }
+            }
         }
 
-        ~SerialStream()
+        private void FinishPendingIORequests(Func<Exception> createException = null)
         {
-            Dispose(false);
+            createException = createException ?? InternalResources.FileNotOpenException;
```

Comment:
This is going to allocate an exception as part of Dispose even if there are no pending read/write requests?

Reply:
this was allocating the `func<exception>` - I'm removing it as part of the other feedback anyway

Reply:
> this was allocating the func<exception>

Ah, yes.